engine.py

```diff
@@ -23,200 +19,240 @@
+        if self.context.project is None:
+            # Must have at least a project in the context to even start!
+            raise tank.TankError("The nuke engine needs at least a project"
+                                 "in the context in order to start! Your "
+                                 "context: %s" % self.context)
         
-        # make sure that nuke has a higher version than 6.3v5
-        # this is because of pyside
-        nuke_version = (nuke.env.get("NukeVersionMajor"), nuke.env.get("NukeVersionMinor"), nuke.env.get("NukeVersionRelease"))
+        # Make sure that nuke has a higher version than 6.3v5.
+        # This is because of PySide.
+        nuke_version = (
+            nuke.env.get("NukeVersionMajor"),
+            nuke.env.get("NukeVersionMinor"),
+            nuke.env.get("NukeVersionRelease")
+        )
         
         if nuke_version[0] < 6:
             self.log_error("Nuke 6.3v5 is the minimum version supported!")
             return
-        elif (nuke_version[0] == 6
-              and nuke_version[1] < 3):
+        elif nuke_version[0] == 6 and nuke_version[1] < 3:
             self.log_error("Nuke 6.3v5 is the minimum version supported!")
             return
-        elif (nuke_version[0] == 6
-              and nuke_version[1] == 3
-              and nuke_version[2] < 5):
+        elif nuke_version[0] == 6 and nuke_version[1] == 3 and nuke_version[2] < 5:
             self.log_error("Nuke 6.3v5 is the minimum version supported!")
             return
-        
-        # keep track of if a UI exists
-        self._ui_enabled = nuke.env.get("gui")
 
-        # versions > 9.0 have not yet been tested so show a message to that effect:
-        if nuke_version[0] > 9 or (nuke_version[0] == 9 and nuke_version[1] > 0):
-            # this is an untested version of Nuke
-            msg = ("The Shotgun Pipeline Toolkit has not yet been fully tested with Nuke %d.%dv%d. "
-                   "You can continue to use the Toolkit but you may experience bugs or "
-                   "instability.  Please report any issues you see to support@shotgunsoftware.com" 
-                   % (nuke_version[0], nuke_version[1], nuke_version[2]))
-            
-            # show nuke message if in UI mode, this is the first time the engine has been started
-            # and the warning dialog isn't overriden by the config:
-            if (self._ui_enabled 
-                and not "TANK_NUKE_ENGINE_INIT_NAME" in os.environ
-                and nuke_version[0] >= self.get_setting("compatibility_dialog_min_version", 10)):
-                nuke.message("Warning - Shotgun Pipeline Toolkit!\n\n%s" % msg)
-                           
-            # and log the warning
-            self.log_warning(msg)
-            
-        # now prepare tank so that it will be picked up by any new processes
+        # Now prepare tank so that it will be picked up by any new processes
```

Comment:
We shouldn't take this bit out. That's something we use to tell people what max version we support! Instead, bump the numbers to the current version we are testing against, so that all future versions will report the warning.